### PR TITLE
[Merged by Bors] - chore: remove one superfluous `nolint defLemma`

### PR DESCRIPTION
**Pull request by grunweg**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/WithZeroTopology.lean [hunk 1]
```diff
@@ -154,7 +154,6 @@ theorem isOpen_Iio {a : Γ₀} : IsOpen (Iio a) :=
 
 /-- The topology on a linearly ordered group with zero element adjoined is compatible with the order
 structure: the set `{p : Γ₀ × Γ₀ | p.1 ≤ p.2}` is closed. -/
-@[nolint defLemma]
 scoped instance (priority := 100) orderClosedTopology : OrderClosedTopology Γ₀ where
```
Comment (grunweg) [inline Mathlib/Topology/Algebra/WithZeroTopology.lean:157]:
```suggestion
@[nolint defLemma]
scoped instance (priority := 100) orderClosedTopology : OrderClosedTopology Γ₀ where

File: Mathlib/Topology/Algebra/WithZeroTopology.lean [hunk 2]
```diff
@@ -176,7 +174,6 @@ scoped instance (priority := 100) t5Space : T5Space Γ₀ where
 
 /-- The topology on a linearly ordered group with zero element adjoined makes it a topological
 monoid. -/
-@[nolint defLemma]
 scoped instance (priority := 100) : ContinuousMul Γ₀ where
```
Comment (grunweg) [inline Mathlib/Topology/Algebra/WithZeroTopology.lean:177]:
```suggestion
@[nolint defLemma]
scoped instance (priority := 100) : ContinuousMul Γ₀ where

File: Mathlib/Topology/Algebra/WithZeroTopology.lean [hunk 3]
```diff
@@ -199,7 +196,6 @@ scoped instance (priority := 100) : ContinuousMul Γ₀ where
       rw [nhds_prod_eq, nhds_of_ne_zero hx, nhds_of_ne_zero hy, prod_pure_pure]
       exact pure_le_nhds (x * y)
 
-@[nolint defLemma]
 scoped instance (priority := 100) : HasContinuousInv₀ Γ₀ :=
```
Comment (grunweg) [inline Mathlib/Topology/Algebra/WithZeroTopology.lean:199]:
```suggestion
@[nolint defLemma]
scoped instance (priority := 100) : HasContinuousInv₀ Γ₀ :=

File: Mathlib/CategoryTheory/Category/Preorder.lean [hunk 4]
```diff
@@ -90,7 +90,7 @@ theorem leOfHom {x y : X} (h : x ⟶ y) : x ≤ y :=
   h.down.down
 #align category_theory.le_of_hom CategoryTheory.leOfHom
 
-@[nolint defLemma, inherit_doc leOfHom]
+@[inherit_doc leOfHom]
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Category/Preorder.lean:93]:
```suggestion
@[nolint defLemma, inherit_doc leOfHom]

File: Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean [hunk 5]
```diff
@@ -28,7 +28,7 @@ namespace CategoryTheory.NormalMonoCategory
 variable [HasFiniteProducts C] [HasKernels C] [NormalMonoCategory C]
 
 /-- The pullback of two monomorphisms exists. -/
-@[irreducible, nolint defLemma] -- Porting note: changed to irreducible and a def
+@[irreducible] -- Porting note: changed to irreducible and a def
 def pullback_of_mono {X Y Z : C} (a : X ⟶ Z) (b : Y ⟶ Z) [Mono a] [Mono b] :
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean:32]:
```suggestion
@[irreducible, nolint defLemma] -- Porting note: changed to irreducible and a def
def pullback_of_mono {X Y Z : C} (a : X ⟶ Z) (b : Y ⟶ Z) [Mono a] [Mono b] :

File: Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean [hunk 6]
```diff
@@ -105,7 +105,7 @@ private abbrev P {X Y : C} (f g : X ⟶ Y) [Mono (prod.lift (𝟙 X) f)] [Mono (
 
 /-- The equalizer of `f` and `g` exists. -/
  -- Porting note: changed to irreducible def since irreducible_def was breaking things
-@[irreducible, nolint defLemma]
+@[irreducible]
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean:108]:
```suggestion
@[irreducible, nolint defLemma]

File: Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean [hunk 7]
```diff
@@ -183,7 +183,7 @@ namespace CategoryTheory.NormalEpiCategory
 variable [HasFiniteCoproducts C] [HasCokernels C] [NormalEpiCategory C]
 
 /-- The pushout of two epimorphisms exists. -/
-@[irreducible, nolint defLemma] -- Porting note: made a def and re-added irreducible
+@[irreducible] -- Porting note: made a def and re-added irreducible
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean:186]:
```suggestion
@[irreducible, nolint defLemma] -- Porting note: made a def and re-added irreducible

File: Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean [hunk 8]
```diff
@@ -28,6 +28,7 @@ namespace CategoryTheory.NormalMonoCategory
 variable [HasFiniteProducts C] [HasKernels C] [NormalMonoCategory C]
 
 /-- The pullback of two monomorphisms exists. -/
+@[irreducible] -- Porting note: changed to irreducible and a def
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean:31]:
```suggestion

File: Mathlib/RingTheory/Kaehler/Basic.lean [hunk 9]
```diff
@@ -416,19 +416,16 @@ local instance smul_SSmod_SSmod : SMul (S ⊗[R] S ⧸ KaehlerDifferential.ideal
     (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2) := Mul.toSMul _
 
 /-- A shortcut instance to prevent timing out. Hopefully to be removed in the future. -/
-@[nolint defLemma]
 local instance isScalarTower_S_right :
```
Comment (grunweg) [inline Mathlib/RingTheory/Kaehler/Basic.lean:419]:
```suggestion
@[nolint defLemma]
local instance isScalarTower_S_right :

File: Mathlib/RingTheory/Kaehler/Basic.lean [hunk 10]
```diff
@@ -416,19 +416,16 @@ local instance smul_SSmod_SSmod : SMul (S ⊗[R] S ⧸ KaehlerDifferential.ideal
...
     IsScalarTower S (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2)
       (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2) := Ideal.Quotient.isScalarTower_right
 
 /-- A shortcut instance to prevent timing out. Hopefully to be removed in the future. -/
-@[nolint defLemma]
 local instance isScalarTower_R_right :
```
Comment (grunweg) [inline Mathlib/RingTheory/Kaehler/Basic.lean:424]:
```suggestion
@[nolint defLemma]
local instance isScalarTower_R_right :

File: Mathlib/RingTheory/Kaehler/Basic.lean [hunk 11]
```diff
@@ -416,19 +416,16 @@ local instance smul_SSmod_SSmod : SMul (S ⊗[R] S ⧸ KaehlerDifferential.ideal
...
     IsScalarTower R (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2)
       (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2) := Ideal.Quotient.isScalarTower_right
 
 /-- A shortcut instance to prevent timing out. Hopefully to be removed in the future. -/
-@[nolint defLemma]
 local instance isScalarTower_SS_right : IsScalarTower (S ⊗[R] S)
```
Comment (grunweg) [inline Mathlib/RingTheory/Kaehler/Basic.lean:429]:
```suggestion
@[nolint defLemma]
local instance isScalarTower_SS_right : IsScalarTower (S ⊗[R] S)

File: Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean [hunk 12]
```diff
@@ -266,7 +266,7 @@ private abbrev Q {X Y : C} (f g : X ⟶ Y) [Epi (coprod.desc (𝟙 Y) f)] [Epi (
   pushout (coprod.desc (𝟙 Y) f) (coprod.desc (𝟙 Y) g)
 
 /-- The coequalizer of `f` and `g` exists. -/
-@[irreducible, nolint defLemma] -- Porting note: changed to def and restored irreducible
+@[irreducible] -- Porting note: changed to def and restored irreducible
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/NormalMono/Equalizers.lean:269]:
```suggestion
@[irreducible, nolint defLemma] -- Porting note: changed to def and restored irreducible

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean [hunk 13]
```diff
@@ -183,7 +183,7 @@ structure BicartesianSq {W X Y Z : C} (f : W ⟶ X) (g : W ⟶ Y) (h : X ⟶ Z)
 #align category_theory.bicartesian_sq CategoryTheory.BicartesianSq
 
 -- Lean should make these parent projections as `lemma`, not `def`.
-attribute [nolint defLemma docBlame] BicartesianSq.toIsPullback BicartesianSq.toIsPushout
+attribute [nolint defLemma docBlame] /-BicartesianSq.toIsPullback-/ BicartesianSq.toIsPushout
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:186]:
```suggestion
attribute [nolint defLemma docBlame] BicartesianSq.toIsPushout

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**